### DISCUSSION TITLE
Is it possible to get the final results faster in the real-time Speech-to Text?

### DISCUSSION THREAD

**Discussion by MiguelHerreroIfemaLab** [deepgram/community#785] [General help] [answered]
I'm working on a subtitles system that overlays real-time transcription on my PC. I'm using QWidget and a QLabel to display the text. My problem is that I want to avoid corrections in the transcription. I've tried using only the interim results, but I get many misspellings without the final results' corrections. Then I tried using only the final results, and it works well, but I'm hoping to reduce the delay. Is there any way to get the final results faster even if i get a higher % or error, or is this just how Deepgram works?

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by nikolawhallon** [comment, marked as answer]
TL;DR This is in some sense indeed just how Deepgram works.

Final results are streamed on average once every 3-5 seconds, except when a speaker stops speaking, we may return a result in that case sooner than 3 seconds. The accuracy of final results goes up when a reasonable amount of audio context is given to the model (hence the 3+ second chunk durations). It's possible to do inferencing on much smaller chunks, but the accuracy would go down (this is what interim results are - just predictions made on smaller chunks). Being able to produce smaller and smaller chunks while maintaining accuracy is a solid research goal, and something we certainly think about, but there's nothing to report on this right now.

**Comment by tarungarg546** [comment]
@nikolawhallon @team-deepgram - is there a way to increase this 3-5 seconds window? We're self hosting deepgram at our end.
